### PR TITLE
Add batch apply filter and check the result of creating layer

### PR DESCRIPTION
**Pull request by lisa0314**
@huningxin , please help to review.

### CODE REVIEW

File: services/ml/compilation_impl_mac.mm [hunk 1]
```diff
@@ -483,8 +483,14 @@ void ComputeBNNSOffsetForImplicitPadding(
       out_desc.row_stride = output_width;
       out_desc.image_stride = output_width * output_height;
       out_desc.data_type = BNNSDataTypeFloat32;
-      operation.filter = BNNSFilterCreateConvolutionLayer(
+      BNNSFilter filter = BNNSFilterCreateConvolutionLayer(
```
Comment (fujunwei) [inline services/ml/compilation_impl_mac.mm]: Why don't check operation.filter?

**Comment by lisa0314** [comment]
@fujunwei Done! Thanks for your review!

**Comment by lisa0314** [comment]
@fujunwei Please help to review again.

File: services/ml/execution_impl_mac.mm [hunk 2]
```diff
@@ -341,7 +343,16 @@ bool GetMPSImageInfo(const OperandMac& operand,
...
-              int result = BNNSFilterApply(operation.filter, src, des);
+              int32_t input_batch_size = operation_input.dimensions[0];
+              int result;
+              if (input_batch_size == 1) {
+                result = BNNSFilterApply(operation.filter, src, des);
+              } else {
+                result = BNNSFilterApplyBatch(operation.filter, input_batch_size, src, 
+                    operation_input.dimensions[1] * operation_input.dimensions[2] * 
+                    operation_input.dimensions[3], des, operation_output.dimensions[1] *
+                    operation_output.dimensions[2] * operation_output.dimensions[3]);
```
Comment (fujunwei) [inline services/ml/execution_impl_mac.mm]: Use "git cl format".

**Comment by fujunwei** [comment]
LGTM

**Pull request merged**